source/postprocess/geoid.cc

```diff
@@ -524,116 +609,114 @@ namespace aspect
-                  // write out the data on processor 0
-                  surface_dynamic_topo_contribution_SH_coes_file << output_surface_dynamic_topo_contribution_SH_coes.str();
-                }
+          // Because each processor already held all the SH coefficients from surface topography contribution,
+          // we only need to stop by the processor 0 to get the data. On processor 0, collect all the data
+          // and put them into the output surface topography contribution SH coefficients file.
+          if (dealii::Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0)
+            {
+              std::ofstream surface_topo_contribution_SH_coes_file (surface_topo_contribution_SH_coes_filename.c_str());
+              surface_topo_contribution_SH_coes_file << "# "
+                                                     << "degree order cosine_coefficient sine_coefficient"
+                                                     << std::endl;
+              std::ostringstream output_surface_delta_rho;
+              output_surface_delta_rho << surface_delta_rho;
+              surface_topo_contribution_SH_coes_file << "surface density contrast(kg/m^3): "
+                                                     << output_surface_delta_rho.str()
+                                                     << std::endl;
+              // Write out the data on processor 0
+              surface_topo_contribution_SH_coes_file << output_surface_topo_contribution_SH_coes.str();
             }
+        }
 
-          // The user can get the spherical harmonic coefficients of the CMB dynamic topography contribution if needed
-          if (also_output_CMB_dynamic_topo_contribution_SH_coes == true)
-            {
-              // have a stream into which we write the SH coefficients data from CMB dynamic topography contribution.
-              // The text stream is then later sent to processor 0
-              std::ostringstream output_CMB_dynamic_topo_contribution_SH_coes;
+      // The user can get the spherical harmonic coefficients of the CMB topography contribution if needed.
+      if (also_output_CMB_topo_contribution_SH_coes == true)
+        {
+          // Have a stream into which we write the SH coefficients data from CMB topography contribution.
+          // The text stream is then later sent to processor 0.
+          std::ostringstream output_CMB_topo_contribution_SH_coes;
 
-              // Prepare the output SH coefficients data from CMB dynamic topography contribution.
-              unsigned int SH_coes_ind = 0;
-              for (unsigned int ideg =  min_degree; ideg < max_degree+1; ideg++)
+          // Prepare the output SH coefficients data from CMB topography contribution.
+          unsigned int SH_coes_ind = 0;
+          for (unsigned int ideg =  min_degree; ideg < max_degree+1; ideg++)
+            {
+              for (unsigned int iord = 0; iord < ideg+1; iord++)
```

Comment:
same comment about changing `x++` into `++x`

Reply:
should all ideg++ be changed to ++ideg?

Reply:
yes all places